engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/ProcessingScheduleServiceTest.java

```diff
@@ -240,13 +242,14 @@ public void shouldScheduleOnFixedRate() {
   void shouldPreserveOrderingOfWritesEvenWithRetries() {
     // given
     final var dummyProcessorSpy = spy(dummyProcessor);
-    final var syncLogStream = spy(streamPlatform.createLogStream("stream", 1));
+    final var syncLogStream = spy(streamPlatform.getLogStream("stream-1"));
     final var logStream = spy(syncLogStream.getAsyncLogStream());
     final var batchWriter = spy(syncLogStream.newLogStreamBatchWriter());
```

Comment:
:question: If spies preserve resources by keeping references, should we also close them? I thought spies were just proxies so it shouldn't matter, but I'm not super familiar with Java proxies.

Reply:
Oh, I see, doing `when()` creates an object :scream: That makes a lot of sense actually, ups :smile: Wouldn't nothing hold a reference to it though? Hm...

Reply:
I think it was a combination of things. We crated a new logstream for that test and created a writer which caused to create a dispatcher etc. This was not closed and memory was also not freed anyway on close before.

Reply:
If you run other tests in the same class in a loop, do you also run out of memory? I think I did, but it might have been only the dispatcher in those cases.

Reply:
I have run now another test in this class for ~ 700 times